--- OpenVPN/openvpn-gui.nsi	2010-12-03 18:38:41.000000000 +0200
+++ nsis-openvpn-2.0.9/openvpn-gui.nsi	2010-12-03 19:05:29.673621527 +0200
@@ -765,8 +765,14 @@
   SectionSetFlags ${SecMYCERT} 0
 !endif
 
+  ; Don't install OpenVPN service by default
+  SectionSetFlags ${SecService} 0
+
   ; Don't install the TAP driver as hiddden as default.
   SectionSetFlags ${SecTAPHidden} 0
+
+  ; Don't include OpenVPN to PATH by default
+  SectionSetFlags ${SecAddPath} 0
 FunctionEnd
 
 Function .onSelChange
