--- OpenVPN/openvpn-gui.nsi	2010-12-03 18:38:41.000000000 +0200
+++ nsis-openvpn-2.0.9/openvpn-gui.nsi	2010-12-03 19:12:55.920210179 +0200
@@ -291,6 +291,11 @@
   File "${HOME}\sample-config\server.${SERV_CONFIG_EXT}"
 !endif
 
+; Extra install hooks included here
+!ifdef EXTRA_INSTALL_NSI
+  !include EXTRA_INSTALL_NSI
+!endif
+
   SetOutPath "$INSTDIR"
   File "${HOME}\install-win32\OpenVPN GUI ReadMe.txt"
 
