--- nsis-openvpn-2.0.9/openvpn-gui.nsi	2010-12-03 17:27:41.688132777 +0200
+++ nsis-openvpn-2.0.9/openvpn-gui.nsi	2010-12-03 18:06:57.143939787 +0200
@@ -106,7 +106,15 @@
 
   !define MUI_WELCOMEPAGE_TITLE "Welcome to the ${PRODUCT_NAME} Setup Wizard"
 
-  !define MUI_WELCOMEPAGE_TEXT "This wizard will guide you through the installation of:\r\n\r\nOpenVPN -  an Open Source VPN package by James Yonan.\r\n\r\nOpenVPN GUI - A Graphical User Interface for OpenVPN by Mathias Sundman\r\n\r\nMy Certificate Wizard - A tool to create a certificate request by Vlada Macek\r\n\r\nNote that the Windows version of OpenVPN will only run on Win 2000, XP, or higher.\r\n\r\n\r\n"
+  !define WELCOMEPAGE_TEXT_OPENVPN "OpenVPN -  an Open Source VPN package by James Yonan.\r\n\r\n"
+  !define WELCOMEPAGE_TEXT_OPENVPNGUI "OpenVPN GUI - A Graphical User Interface for OpenVPN by Mathias Sundman\r\n\r\n"
+!ifdef SKIP_MYCERT
+  !define WELCOMEPAGE_TEXT_MYCERT ""
+!else
+  !define WELCOMEPAGE_TEXT_MYCERT "My Certificate Wizard - A tool to create a certificate request by Vlada Macek\r\n\r\n"
+!endif
+
+  !define MUI_WELCOMEPAGE_TEXT "This wizard will guide you through the installation of:\r\n\r\n${WELCOMEPAGE_TEXT_OPENVPN}${WELCOMEPAGE_TEXT_OPENVPNGUI}${WELCOMEPAGE_TEXT_MYCERT}Note that the Windows version of OpenVPN will only run on Win 2000, XP, or higher.\r\n\r\n\r\n"
 
   !define MUI_COMPONENTSPAGE_TEXT_TOP "Select the components to install/upgrade.  Stop any OpenVPN or OpenVPN GUI processes or the OpenVPN service if it is running."
 
@@ -231,6 +239,7 @@
 
 SectionEnd
 
+!ifndef SKIP_EASYRSA
 Section "OpenVPN RSA Certificate Management Scripts" SecOpenVPNEasyRSA
 
   SetOverwrite on
@@ -253,6 +262,7 @@
   File "${HOME}\easy-rsa\serial.start"
 
 SectionEnd
+!endif
 
 Section "OpenVPN GUI" SecGUI
 
@@ -261,13 +271,15 @@
   File "${HOME}\openvpn-gui.exe"
 
   # Include your custom config file(s) here.
-  SetOutPath "$INSTDIR\config"
+;  SetOutPath "$INSTDIR\config"
   ;File "${HOME}\config\Office.ovpn"
 
+!ifndef SKIP_SAMPLE_CONFIG
   SetOutPath "$INSTDIR\sample-config"
   File "${HOME}\sample-config\sample.${SERV_CONFIG_EXT}"
   File "${HOME}\sample-config\client.${SERV_CONFIG_EXT}"
   File "${HOME}\sample-config\server.${SERV_CONFIG_EXT}"
+!endif
 
   SetOutPath "$INSTDIR"
   File "${HOME}\install-win32\OpenVPN GUI ReadMe.txt"
@@ -280,6 +292,7 @@
 Section "AutoStart OpenVPN GUI" SecGUIAuto
 SectionEnd
 
+!ifndef SKIP_MYCERT
 Section "My Certificate Wizard" SecMYCERT
 
   SetOverwrite on
@@ -288,6 +301,7 @@
   File "${HOME}\mycertwizard\mycert.ini"
 
 SectionEnd
+!endif
 
 Section "Hide the TAP-Win32 Virtual Ethernet Adapter" SecTAPHidden
 SectionEnd
@@ -307,11 +321,12 @@
   FileWrite $R0 "process will be instantiated for each configuration file.$\r$\n"
   FileClose $R0
 
+!ifndef SKIP_SAMPLE_CONFIG
   SetOutPath "$INSTDIR\sample-config"
   File "${HOME}\sample-config\sample.${SERV_CONFIG_EXT}"
   File "${HOME}\sample-config\client.${SERV_CONFIG_EXT}"
   File "${HOME}\sample-config\server.${SERV_CONFIG_EXT}"
-
+!endif
 
   CreateDirectory "$INSTDIR\log"
   FileOpen $R0 "$INSTDIR\log\README.txt" w
@@ -324,6 +339,7 @@
 Section "OpenVPN File Associations" SecFileAssociation
 SectionEnd
 
+!ifndef SKIP_OPENSSL
 Section "OpenSSL DLLs" SecOpenSSLDLLs
 
   SetOverwrite on
@@ -344,6 +360,7 @@
   File "${BIN}\libssl32.dll"
 
 SectionEnd
+!endif
 
 Section "OpenSSL Utilities" SecOpenSSLUtilities
 
@@ -643,8 +660,10 @@
       CreateShortCut "$SMPROGRAMS\OpenVPN\OpenVPN log file directory.lnk" "$INSTDIR\log" ""
 
  samp_shortcut:
+!ifndef SKIP_SAMPLE_CONFIG
     IfFileExists "$INSTDIR\sample-config" "" genkey_shortcut
       CreateShortCut "$SMPROGRAMS\OpenVPN\OpenVPN Sample Configuration Files.lnk" "$INSTDIR\sample-config" ""
+!endif
 
  genkey_shortcut:
     IfFileExists "$INSTDIR\bin\openvpn.exe" "" noshortcuts
